### PR TITLE
[4.3][csp] Reduce inline onchange usage [searchtools]

### PR DESCRIPTION
**Pull request by dgrammatiko**
Pull Request for Issue # .

### Summary of Changes
- remove the filter xml entries `onchange="this.form.submit();"`
- add a class `js-select-submit-on-change` on their place
- Tweet the search tools javascript to replicate the functionality for each element with the class

### Testing Instructions
This PR needs `npm ci`

Pick a list view and check that all the filters are working as before (ie selecting anything with a select element and observe the page refreshing)

### Actual result BEFORE applying this Pull Request

Works

### Expected result AFTER applying this Pull Request

Works

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

**Comment by wilsonge** [comment]
I think this is fine. But we are definitely going to need to document all these magic classes if we expect extensions to adopt this as part of CSP implementation of the backend.

**Comment by dgrammatiko** [comment]
@wilsonge is there a section on the docs fro the search tools? If not can someone add it?

File: administrator/components/com_privacy/forms/filter_consents.xml [hunk 1]
```diff
@@ -70,7 +70,7 @@
 			label="JGLOBAL_LIST_LIMIT"
 			class="input-mini"
 			default="25"
-			onchange="this.form.submit();"
+			class="js-select-submit-on-change"
```
Comment (Quy) [inline administrator/components/com_privacy/forms/filter_consents.xml]:
```suggestion
			class="input-mini js-select-submit-on-change"
```

File: administrator/components/com_privacy/forms/filter_consents.xml [hunk 2]
```diff
@@ -70,7 +70,7 @@
 			label="JGLOBAL_LIST_LIMIT"
 			class="input-mini"
```
Comment (Quy) [inline administrator/components/com_privacy/forms/filter_consents.xml]:
```suggestion
```

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 08e9766e55890ffcf996ff3a0a16315d0c49dc2f<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39726">issues.joomla.org/tracker/joomla-cms/39726</a>.</sub>

**Comment by viocassel** [comment]
I have tested this item :white_check_mark: successfully on 08e9766e55890ffcf996ff3a0a16315d0c49dc2f<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39726">issues.joomla.org/tracker/joomla-cms/39726</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39726">issues.joomla.org/tracker/joomla-cms/39726</a>.</sub>

**Comment by obuisard** [comment]
Thank you Dimitris @dgrammatiko !

**Comment by Quy** [comment]
@dgrammatiko Can you please look into Menus > Manage and toggle Site/Administrator dropdown not updating? Thanks.

**Comment by dgrammatiko** [comment]
> @dgrammatiko Can you please look into Menus > Manage and toggle Site/Administrator dropdown not updating? Thanks.

Will do a PR later on today, thanks for reporting it

**Pull request merged**